app/templates/newmaps.html

```diff
@@ -8,62 +9,156 @@
+                      myLat = newMarker.lat();
+                      myLng = newMarker.lng();
+                      markers.push({
+                        lat : newMarker.lat(),
+                        lng : newMarker.lng(),
+                        title: "Current Address"
+                      }); 
+                    }
+                    map.addMarkers(markers);  
+                    console.log(markers);    
+                    map.fitZoom();  
+                  }
+                  });
           // Provide helpful error message to user.
           // Clear any errors.
+
+          if (Number.isInteger(zipCode)) {
           $("#map-search-error")
             .html("<small class=error>Zip code must consist of exactly 5 digits (0-9).</small>");
-            return;
+            return; }
+          
+
         }
         // Store search result in database
         var json_data = {
           zipCode: zipCode
         };
-        $.post(url = $SCRIPT_ROOT + '/_search_query', data = json_data);
-
-
+        //cases for different checkboxes
+        var queryString = 'search_query'
+        if ($("#map-filter-by-seniors").is(':checked')) { queryString += '_seniors'; }
+        if ($("#map-filter-by-seniors").is(':checked')) { queryString += '_family';}
+        {
+          queryString += "_seniors";
+        }
+        if ($("#map-filter-by-seniors").is(':checked'))
+        {
+          queryString += "_seniors";
+        }
+        if ($("#map-filter-by-seniors").is(':checked'))
+        {
+          queryString += "_seniors";
+        }                        
+        $.post(url = $SCRIPT_ROOT + '/_search_query_zipcode', data = json_data);
         GMaps.geocode({
           address: $('#address').val().trim(),
           callback: function(results, status){
             if(status=='OK'){
-              map.removeMarkers();
               var markers = [];
               var latlng = results[0].geometry.location;
               map.setCenter(latlng.lat(), latlng.lng());
-              var zipCode = results[0].address_components[0].short_name;
-                  var minLat = latlng.lat();
-                  var maxLat = latlng.lat();
-                  var minLng = latlng.lng();
-                  var maxLng = latlng.lng();              
+              var zipCode = results[0].address_components[0].short_name;            
             $.getJSON($SCRIPT_ROOT + '/_map', {
           zip_code: zipCode
         },
             function(data) {
+              
+              for (var j = data["addresses"].length; j >=0; j--)
+              {
+                var food_resource_id = data["addresses"][j]["food_resource_id"];
+                if (($("#map-filter-by-seniors").is(':checked')) && 
+                  data["addresses"][j]["food_resource_id"]          )
+              }
+              if ($("#map-filter-by-seniors").is(':checked')) //filtering results by checkbox categories
```

Comment:
Is this checkbox code currently functional? If not, can you comment it out and explain what it's current behavior is.